cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -422,6 +443,15 @@ func (c *machineController) machineSetProviderIDs(machineSet *MachineSet) ([]str
 			continue
 		}
 
+		if machine.Status.FailureMessage != nil {
+			klog.V(4).Infof("Status.FailureMessage of machine %q is %q", machine.Name, *machine.Status.FailureMessage)
+			// Provide a fake ID that can be recognised later and converted into a machine key.
```

Comment:
nit: may be add the "why" i.e so the machine is tracked by the nodegroup and the maxNodeProvisionTime can kicks in